apps/frontend/src/app/Formula/internal.ts

```diff
@@ -88,7 +88,7 @@ export function customMapFormula<Context, Output, Input extends Base<Input>>(for
     const [visiting, mapping] = current
 
     const old = mapping.get(formula)
-    if (old) return old
+    if (old) return { ...old }

```

Comment:
Don't create new objects; it'll decache the results.

Reply:
Oh lmao I did this cuz I found the cached objects were being modified, breaking the bounding alg. How can I change customMapFormula to return a read-only object?

Reply:
We unfortunately does not have guardrails to prevent mutation, but immutability is the foundational assumption of waverider. `customMapFormula` itself does not mutate a `Node`, nor do any other existing functions. There are only a handful of exceptions, but they are limited to when it is guaranteed that the mutation is not observable.

Make sure there is no mutation in your code. If the mutation comes from existing codebase, please file a bug report.